src/librustc_typeck/diagnostics.rs

```diff
@@ -4163,6 +4163,34 @@ target / ABI combination is currently unsupported by llvm.
 If necessary, you can circumvent this check using custom target specifications.
 "##,
 
+E0571: r##"
+A return statement was outside a function scope.
+
+Erroneous code example:
+
+```compile_fail,E0571
+const FOO: u32 = return 0; // error: return statement cannot be out of a
+                           //        function scope
+
+fn main() {}
+```
+
+To fix this issue, just remove the return statement or move it into a function
```

Comment:
Considering the statement/expression grammatical categories, return is an expression [according to the reference](https://doc.rust-lang.org/reference.html#return-expressions). So to say "remove the return statement" is imprecise because it's an expression, and misleading because the user might try to remove the entire expression when he should only remove the `return` keyword.